### PR TITLE
Homework4: Alexandra Chigir

### PR DESCRIPTION
**Pull request by as-chigir**
### PR checklist
Please make sure you've passed PR checklist
- [ ] my PR passes CI checks (flake8 and pycodestyle)
- [ ] my PR named according to guidelines.
- [ ] my PR contains complete tasks descriptions for all tasks
- [ ] my PR made from the fork feature branch to maxim homework# branch.
- [ ] my PR contains only code that solves appropriate homework tasks. Redundant 
files (configs, etc. ) are put to .gitignore

### CODE REVIEW

File: src/Homework4/task2.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+
+n = int(input('Введите кол-во стран: '))
+for i in range(n):
+    capitals = input('Введите ' + str(i + 1) + ' -ю страну и ее города: ')
+    a = capitals.split()
+m = int(input('Введите кол-во городов: '))
+for j in range(m):
+    cities = input('Введите ' + str(j + 1) + ' -й город: ')
+    for k in cities:  # ИСПРАВИТЬ !!! не работает
```
Comment (belovmd) [inline src/Homework4/task2.py]: comments

File: src/Homework4/task2.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        if cities in a:
+            print(a[0])
+
+
+"""N = int(input('Введите кол-во стран: '))
```
Comment (belovmd) [inline src/Homework4/task2.py]: strange comments

File: src/Homework4/task2.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+   Moscow
+   Novgorod
+   Выходные данные
+   Ukraine
+   Russia
+   Russia
+"""
+
+
+n = int(input('Введите кол-во стран: '))
```
Comment (belovmd) [inline src/Homework4/task2.py:34]: contradicts task description input

File: src/Homework4/task3.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"""Даны два списка чисел. Посчитайте, сколько различных чисел содержится
+   одновременно как в первом списке, так и во втором.
+"""
+
+
```
Comment (belovmd) [inline src/Homework4/task3.py:6]: it's a task for sets operations

File: src/Homework4/task4.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+"""Даны два списка чисел. Посчитайте, сколько различных чисел входит только
+   в один из этих списков
+"""
+
```
Comment (belovmd) [inline src/Homework4/task4.py:4]: task for sets operations

File: src/Homework4/task5.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+for i in range(n):
+    num_lang = int(input('Введите кол-во языков ' + str(i + 1) + '-го шк-ка:'))
+    for j in range(num_lang):
+        lang = input('Введите язык школьника: ')
+        studs[lang] = num_lang
+    s += num_lang
+    total_lang = s
+
+print('\nКол-во языков, которые знают все школьники: ', total_lang)
+# Вывести языки всех школьников !!! НЕ СДЕЛАНО
```
Comment (belovmd) [inline src/Homework4/task5.py]: not done

File: src/Homework4/task5.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+   Выходные данные
+   В первой строке выведите количество языков, которые знают все школьники.
+   Начиная со второй строки - список таких языков. Затем - количество языков,
+   которые знает хотя бы один школьник, на следующих строках - список таких
+   языков.
+"""
+
+studs = {}
+n = int(input("Введите кол-во школьников: "))
+s = 0
```
Comment (belovmd) [inline src/Homework4/task5.py:36]: unclear naming

File: src/Homework4/task5.py [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+for i in range(n):
+    num_lang = int(input('Введите кол-во языков ' + str(i + 1) + '-го шк-ка:'))
+    for j in range(num_lang):
+        lang = input('Введите язык школьника: ')
+        studs[lang] = num_lang
+    s += num_lang
+    total_lang = s
+
+print('\nКол-во языков, которые знают все школьники: ', total_lang)
```
Comment (belovmd) [inline src/Homework4/task5.py]: doesn't solve task in description

File: src/Homework4/task6.py [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+   слов содержится в этом тексте.
+"""
+
+
+s1 = 'I ? will definitely . learn Python !'
+s2 = ['.', '!', '?']
+c = 0
+
+for i in s1.split():
+    if i not in s2:
```
Comment (belovmd) [inline src/Homework4/task6.py]: has one line solution

File: src/Homework4/task7.py [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+
+a = 10
+b = 4
+
+while a > b:
+    if a - b == 0:
+        print(a)
+    else:
+        a, b == b, a - b
```
Comment (belovmd) [inline src/Homework4/task7.py]: where is result?

**Review by belovmd — CHANGES_REQUESTED** [review]

File: src/Homework4/task2.py (hunk 1 above)
Comment (as-chigir) [reply src/Homework4/task2.py]: yes, I have not yet understood how))) this is not to forget

File: src/Homework4/task2.py [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+   Russia
+"""
+
+
+capitals_cities = {}
+n = int(input('Введите кол-во стран: '))
+for i in range(n):
+    capitals = input('Введите ' + str(i + 1) + ' -ю страну и ее города: ')
+    a = capitals.split()
+    capital = a[0]
```
Comment (belovmd) [inline src/Homework4/task2.py:38]: country, *cities = str_.split()

Comment (belovmd) [reply src/Homework4/task2.py:38]: unclear naming

File: src/Homework4/task2.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    cities = a[1:]
+    for k in cities:
+        capitals_cities[k] = capital
+
+m = int(input('Введите кол-во городов: '))
+for j in range(m):
+    city = input('Введите ' + str(j + 1) + ' -й город: ')
```
Comment (belovmd) [inline src/Homework4/task2.py:45]: doesn't work if Brest in France and Belarus

File: src/Homework4/task3.py [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+   чисел содержится одновременно как в первом списке,
+   так и во втором.
+"""
+
+
+s1 = [1, 2, 1, 1, 3, 4, 5, 8, 9]
+s2 = [10, 10, 11, 1, 2, 6, 7]
+a = set(s1)
+b = set(s2)
+print(len(a.union(b)))
```
Comment (belovmd) [inline src/Homework4/task3.py:11]: doesn't solve task in description

File: src/Homework4/task4.py [hunk 14]
```diff
@@ -0,0 +1,10 @@
+"""Даны два списка чисел. Посчитайте, сколько различных
+   чисел входит только в один из этих списков.
+"""
+
+
+s1 = [1, 2, 1, 1, 3, 4, 5, 8, 9]
+s2 = [10, 10, 11, 1, 2, 6, 7]
+a = set(s1)
+b = set(s2)
+print(len(a.difference(b)))
```
Comment (belovmd) [inline src/Homework4/task4.py:10]: doesn't solve task in description

File: src/Homework4/task5.py (hunk 7 above)
Comment (belovmd) [reply src/Homework4/task5.py:36]: not addressed

File: src/Homework4/task5.py [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+    num_lang = int(input('К-во языков ' + str(i + 1) + '-го шк-ка:'))
+    for j in range(num_lang):
+        lang = input('Введите язык школьника: ')
+        studs[lang] = num_lang
+        list_lang.append(lang)
+    s += num_lang
+    total_lang = s
+
+print('\nК-во языков, которые знают все школьники: ', total_lang)
+for k in list_lang:
```
Comment (belovmd) [inline src/Homework4/task5.py:47]: doesn't solve task in description

**Review by belovmd — APPROVED** [review]

**Comment by belovmd** [comment]
6

**Pull request closed without merging**